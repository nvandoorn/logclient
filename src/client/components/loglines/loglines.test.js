/* eslint-env jest */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Loglines from './loglines'

const props = [{
  text: `You must be pretty desperate if you're looking at the logs`,
  level: 2,
  levelStr: 'info',
  datetime: 1489387840000,
  datetimeStr: '6:50:40'
}]

it('should render correctly', () => {
  const loglines = renderer.create(<Loglines loglines={props} />).toJSON()
  expect(loglines).toMatchSnapshot()
})

it('should render without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Loglines loglines={props} />, div)
})
