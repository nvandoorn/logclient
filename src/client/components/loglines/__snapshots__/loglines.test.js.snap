exports[`test renders correctly 1`] = `
<ul
  className={
    Object {
      "list-style-type": "none",
      "margin": 0,
      "padding": 0,
    }
  }>
  <li
    className={
      Object {
        "display": "block",
        "margin": "0 0 20px 0",
      }
    }>
    <div
      className={
        Object {
          "display": "inline-block",
          "font-size": "14px",
          "margin-right": "20px",
        }
      }>
      <div
        className="datetime">
        6:50:40
      </div>
      <div
        className={
          Object {
            "color": "#58B3FC",
            "font-weight": 0,
          }
        }>
        info
      </div>
    </div>
    <div
      className={
        Object {
          "display": "inline-block",
          "vertical-align": "top",
        }
      }>
      You must be pretty desperate if you\'re looking at the logs
    </div>
  </li>
</ul>
`;
